docs/src/man/metadata.md

```diff
@@ -0,0 +1,206 @@
+# Metadata
```

Comment:
@nalimilan - could you please review this file only. I want to settle on the intention before we go to implementation of these rules.